### ISSUE TITLE
Cannot update kosmtik 

### ISSUE THREAD

**Issue by cmyk** [kosmtik/kosmtik#190]
I cannot update kosmtik.
Here's the output

```npm ERR! Linux 2.6.32-40-pve
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "-g" "install" "kosmtik"
npm ERR! node v7.2.0
npm ERR! npm  v3.10.9
npm ERR! path /home/osm/npm-global/lib/node_modules/.staging/abbrev-78d92735
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/npm-global/lib/node_modules/.staging/abbrev-78d92735' -> '/home/osm/npm-global/lib/node_modules/kosmtik/node_modules/abbrev'
npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/npm-global/lib/node_modules/.staging/abbrev-78d92735' -> '/home/osm/npm-global/lib/node_modules/kosmtik/node_modules/abbrev'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /home/osm/carto/kosmtik/npm-debug.log
npm ERR! code 1
```

[npm-debug.txt](https://github.com/kosmtik/kosmtik/files/618435/npm-debug.txt)

**Comment by yohanboniface** [comment]
Not sure node-mapnik is ready for Node 7.0.x.
Also, try to `rm -rf node_modules/` before updating, which is a sane habit with npm in general.

**Comment by cmyk** [comment]
Same with v6.9.1

```
npm ERR! Linux 2.6.32-40-pve
npm ERR! argv "/home/osm/.nvm/versions/node/v6.9.1/bin/node" "/home/osm/.nvm/versions/node/v6.9.1/bin/npm" "-g" "install" "kosmtik"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! path /home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-4c7943d8
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-4c7943d8' -> '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/kosmtik/node_modules/abbrev'
npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-4c7943d8' -> '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/kosmtik/node_modules/abbrev'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /home/osm/carto/kosmtik/npm-debug.log
npm ERR! code 1
```

**Comment by cmyk** [comment]
I also removed the node_modules dir.
Doesn't help.

**Comment by cmyk** [comment]
It's weird.
I just tried to install mapnik 3.5.14 without -g option, i.e. locally. And it worked.
When I try to install it globally I get this error message:
```
osm@map:~/carto/kosmtik$ npm install -g mapnik

> mapnik@3.5.14 preinstall /home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/mapnik-722a003d
> npm install node-pre-gyp

/home/osm/.nvm/versions/node/v6.9.1/bin/node-pre-gyp -> /home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/node-pre-gyp/bin/node-pre-gyp
/home/osm/.nvm/versions/node/v6.9.1/lib
└── node-pre-gyp@0.6.32 

/home/osm/.nvm/versions/node/v6.9.1/lib
└── (empty)

npm ERR! Linux 2.6.32-40-pve
npm ERR! argv "/home/osm/.nvm/versions/node/v6.9.1/bin/node" "/home/osm/.nvm/versions/node/v6.9.1/bin/npm" "install" "-g" "mapnik"
npm ERR! node v6.9.1
npm ERR! npm  v3.10.8
npm ERR! path /home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-7057f722
npm ERR! code ENOENT
npm ERR! errno -2
npm ERR! syscall rename

npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-7057f722' -> '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/mapnik/node_modules/abbrev'
npm ERR! enoent ENOENT: no such file or directory, rename '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/.staging/abbrev-7057f722' -> '/home/osm/.nvm/versions/node/v6.9.1/lib/node_modules/mapnik/node_modules/abbrev'
npm ERR! enoent This is most likely not a problem with npm itself
npm ERR! enoent and is related to npm not being able to find a file.
npm ERR! enoent 

npm ERR! Please include the following file with any support request:
npm ERR!     /home/osm/carto/kosmtik/npm-debug.log
npm ERR! code 1
```
I can also install abbrev globally no problem.

**Comment by cmyk** [comment]
Solved. I had to use node 4.3.0.
Is that mentioned somewhere?

**Comment by nebulon42** [comment]
Should be solved by #194 and in the README it is now mentioned that you should use the LTS version of node. I'm not sure what causes differences when installing node-mapnik globally. I have never tried. Why do you want to do that?

I'm closing this for now. We can re-open if there are additional things to consider.
